src/drawer/pages/ServiceDetail/ServiceDetail.tsx

```diff
@@ -54,12 +57,24 @@ const Category = {
 export const ServiceDetail = () => {
   const { serviceId } = useParams();
   const { product } = useGetProductDetail(Number(serviceId));
+  const productUrls = [
+    { url: product.webpageUrl, text: '웹페이지로 이동' },
+    { url: product.appStoreUrl, text: 'App store로 다운로드' },
+    { url: product.googlePlayUrl, text: 'Google play로 다운로드' },
+    { url: product.githubUrl, text: 'GitHub' },
+  ];
   const queryClient = useQueryClient();
 
   const theme = useTheme();
 
   const scrollRef = useRef<HTMLDivElement | null>(null);
 
+  const { showToast, isShowToast } = useToast();
+  const toastProps = {
+    children: 'URL이 복사되었습니다',
+    duration: 'short' as ToastDuration,
+  };
+
```

Comment:
먼 과거에 읽었던 이펙티브 타입스크립트에 타입 단언을 최대한 피하라는 말이 있었던 것 같은데 타입 단언보다는 [`const assertion`](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-4.html#const-assertions)을 쓰면 어떨까요

```suggestion
  const toastProps = {
    children: 'URL이 복사되었습니다',
    duration: 'short',
  } as const;
```

Reply:
책을 같이 읽어놓고 왜 단언 썼냐: toastProps는 사용자에게 제어되는 값이 아니며, 코드에 박힌 값이라서 변경될 여지가 없으니 단언해도 ㄱㅊ다고 생각함

근데 단언하려면 오직 이것만을 위해 ToastDuration을 import 해야 하는 점이 구리네요
수정 완~~~~!

Reply:
런타임에 타입은 사라지니까 값이 사용자에게 제어되는지 여부는 타입 단언 사용의 이유가 될 수는 없을 것 같습니다🤔

타입스크립트가 선택적 프로퍼티는 무조건 `undefined`를 포함해서 추론해버리니 처리하기가 어려워지긴 하네요

Reply:
코드 작성 시점을 기준으로 말한 거였어요! 만약 사용자가 제어하는 값에 타입 단언을 사용하면 잘못된 값이 들어갈 여지가 있음에도 타입체커가 오류를 무시하니까 발견할 수 없겠지만, short는 그런 경우가 아닌 점을 얘기하려던 거였습니당~